sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/BasicStatsPlanVisitor.scala

```diff
@@ -57,7 +57,15 @@ object BasicStatsPlanVisitor extends LogicalPlanVisitor[Statistics] {
 
   override def visitGlobalLimit(p: GlobalLimit): Statistics = fallback(p)
 
-  override def visitIntersect(p: Intersect): Statistics = fallback(p)
+  override def visitIntersect(p: Intersect): Statistics = {
+    val stats = p.children.map(_.stats)
+    val rowCount = if (stats.exists(_.rowCount.isEmpty)) {
+      None
+    } else {
+      Some(stats.map(_.rowCount.get).min)
+    }
+    Statistics(sizeInBytes = stats.map(_.sizeInBytes).min, rowCount = rowCount)
```

Comment:
We should avoid change the logic, maybe it should be:
```scala
val leftStats = p.left.stats
val rightStats = p.right.stats
val leftSize = leftStats.sizeInBytes
val rightSize = rightStats.sizeInBytes
if (leftSize < rightSize) {
  Statistics(sizeInBytes = leftSize, rowCount = leftStats.rowCount)
} else {
  Statistics(sizeInBytes = rightSize, rowCount = rightStats.rowCount)
}
```

Reply:
yea, done